src/components/Video/Video.component.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useContext } from 'react';
+import { useParams } from 'react-router-dom';
+import useVideoInfo from '../../hooks/useVideoInfo';
+import Styled from './styled';
+import { AppContext } from '../../context/AppProvider';
+
+const Video = () => {
+  const params = useParams();
+  const { videoId } = params;
+  const [title, description] = useVideoInfo(videoId);
+  const videoSrc = `https://www.youtube.com/embed/${videoId}`;
+  const { state, dispatch } = useContext(AppContext);
+  const { isLogged, favourites } = state;
+
+  const addToFavourites = () => {
+    dispatch({ type: 'ADD_FAVOURITE', payload: videoId });
+  };
+
+  const removeFromFavourites = () => {
+    dispatch({ type: 'REMOVE_FAVOURITE', payload: videoId });
+  };
+
+  const favButton = () => {
+    if (favourites.includes(videoId))
+      return (
+        <Styled.Button onClick={removeFromFavourites} data-testid="remFavButton">
+          remove from favourites
+        </Styled.Button>
+      );
+
+    return (
+      <Styled.Button onClick={addToFavourites} data-testid="FavButton">
+        add to favourites
+      </Styled.Button>
+    );
+  };

```

Comment:
This should be place into its own component file. Also, add the curly brackets to the `if` as it is confusing seeing a multiple line return without the brackets.